guides/source/asset_pipeline.md

```diff
@@ -8,479 +8,616 @@ This guide covers the asset pipeline.
-
-When a filename is unique and based on its content, HTTP headers can be set to
-encourage caches everywhere (whether at CDNs, at ISPs, in networking equipment,
-or in web browsers) to keep their own copy of the content. When the content is
-updated, the fingerprint will change. This will cause the remote clients to
-request a new copy of the content. This is generally known as _cache busting_.
+Evolution of Asset Management Techniques
+----------------------------------------
 
-The technique Sprockets uses for fingerprinting is to insert a hash of the
-content into the name, usually at the end. For example a CSS file `global.css`
-
-```
-global-908e25f4bf641868d8683022a5b62f54.css
-```
-
-This is the strategy adopted by the Rails asset pipeline.
-
-Fingerprinting is enabled by default for both the development and production
-environments. You can enable or disable it in your configuration through the
-[`config.assets.digest`][] option.
-
-### What are Import Maps and Why Should I Care?
-
-Import maps let you import JavaScript modules using logical names that map to versioned/digested files – directly from the browser. So you can build modern JavaScript applications using JavaScript libraries made for ES modules (ESM) without the need for transpiling or bundling.
-
-With this approach, you'll ship many small JavaScript files instead of one big JavaScript file. Thanks to HTTP/2 that no longer carries a material performance penalty during the initial transport, and in fact offers substantial benefits over the long run due to better caching dynamics.
-
-How to use Import Maps as JavaScript Asset Pipeline
------------------------------
-
-Import Maps are the default JavaScript processor, the logic of generating import maps is handled by the [`importmap-rails`](https://github.com/rails/importmap-rails) gem.
+Within the last few years, the evolution of the web has led to significant
+changes that have influenced how assets are managed in web applications. These
+include:
 
-WARNING: Import maps are used only for JavaScript files and can not be used for CSS delivery. Check the [Sprockets section](#how-to-use-sprockets) to learn about CSS.
+1. **Browser Support**: Modern browsers have improved support for new features
+   and syntax, reducing the need for transpilation and polyfills.
+2. **HTTP/2**: The introduction of HTTP/2 has made it easier to serve multiple
+   files in parallel, reducing the need for bundling.
+3. **ES6+**: Modern JavaScript syntax (ES6+) is supported by most modern
+   browsers, reducing the need for transpilation and minification.
 
-You can find detailed usage instructions on the Gem homepage, but it's important to understand the basics of `importmap-rails`.
+Therefore, in Rails 8, the asset pipeline powered by Propshaft, no longer
+includes transpilation, bundling, or compression by default. However,
+fingerprinting remains an integral part of Propshaft. You can learn more about
+it below.
```

Comment:
Nice. I like this list and spelling out what Propshaft focuses on.

Reply:
Interesting that you say this @bhumi1102, as my initial reaction was: Why go through everything that the asset pipeline _doesn't_ need before we say what _is_ needed. 

My gut is saying this should be pushed to the bottom as an 'interesting to note' as opposed to treated as a main chunk of the guide. I can see how this would help with SEO when people search, though.

Reply:
Haha, what I meant was I liked the list 1, 2, 3 above. Agree about including what Propshaft _does_ do first (and _not_ focusing what it doesn't need to do).

Reply:
I decided to move the "Evolution of Asset Management Techniques" section under "Sprockets to Propshaft," as it felt more fitting there. A lot of the decisions around moving from Sprockets to Propshaft are tied to how asset management techniques have evolved.